### PR TITLE
Fix makefile build/run under Mac

### PR DESCRIPTION
**Pull request by svanoort**
This is due to a subtle difference between the Linux and Mac 'date' command version which breaks the makefile.  This fixes it.

It also fixes explicitly setting an ip address that may not match for docker-machine hosts vs. native docker.

@reviewbybees esp @jglick

### CODE REVIEW

File: demo/Makefile [hunk 1]
```diff
@@ -1,6 +1,6 @@
-TAG=$(shell date -I -u)
+TAG=$(shell date -u +"%Y-%m-%dT%H-%M-%S")
```
Comment (jglick) [inline demo/Makefile]: No, this produces e.g.

    2017-01-12T19-11-56

whereas I wanted e.g.

    2017-01-12

File: demo/Makefile [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
 IMAGE=jenkinsci/workflow-demo
-DOCKER_RUN=docker run --rm -p 127.0.0.1:2222:2222 -p 127.0.0.1:8080:8080 -p 127.0.0.1:8081:8081 -p 127.0.0.1:9418:9418 -ti
+DOCKER_RUN=docker run --rm -p 2222:2222 -p 8080:8080 -p 8081:8081 -p 9418:9418 -ti
```
Comment (jglick) [inline demo/Makefile]: No, this would make the demo unsafe to run on public WiFi.

Comment (svanoort) [inline demo/Makefile]: @jglick How do you propose to make the demo runnable on Mac, then?

File: demo/Makefile (hunk 1 above)
Comment (svanoort) [inline demo/Makefile]: Not sure why that specific format for the date, but I changed it

File: demo/Makefile (hunk 2 above)
Comment (svanoort) [inline demo/Makefile]: Added a solution - not the prettiest but it will get the job done.

**Comment by svanoort** [comment]
Ugh, nevermind, the docker info still doesn't distinguish between the local and remote servers for IP. 

I can't find a good way to detect a local vs. a docker-machine docker server.

**Comment by svanoort** [comment]
Closing this because I'm pretty sure there's not a way to make this handle docker-machine vs. native correctly.

**Pull request closed without merging**